lib/sax.js

```diff
@@ -576,18 +576,19 @@ function parseDCC(source, start, domBuilder, errorHandler) {
 }
 
 function parseInstruction(source, start, domBuilder) {
-	var end = source.indexOf('?>', start);
```

Comment:
I think this approach is better because `Char*` is greedy, so it would pick as many things as possible, so if there are multiple processing instructions we would only have a singel match. At least this is my current understanding.

And do we also need to prefix `PI_match` with `^` to say that we expect the string to start with it?

Reply:
I already fixed that with Char*?

`var PI_match = reg(/^<\?/, '(', Name, ')', regg(S, '(', Char, '*?)'), '?', /\?>/);`

Reply:
I will double check what ES version is required for this to work, great catch.

Reply:
> And do we also need to prefix `PI_match` with `^` to say that we expect the string to start with it?

Adding `^` here increases performance (for example if regex is `/^<\?/` and input is `<!.....` the regex can return after checking just 2 chars, otherwise it would scan the entire source to find the first <?).
Then we see it is also crucial, because the parser always expects the found match to start at the beginning of the string.
Imagine there are two PI, the first one is invalid, the second one valid, without `^` we would match the second PI, breaking the parser.